test/integration/apiserver/oidc/oidc_test.go

```diff
@@ -348,7 +382,25 @@ func configureTestInfrastructure(t *testing.T) (
 	require.NoError(t, err)
 
 	oidcServer = utilsoidc.BuildAndRunTestServer(t, caFilePath, caKeyFilePath)
-	apiServer = startTestAPIServerForOIDC(t, oidcServer.URL(), defaultOIDCClientID, caFilePath)
+
+	var authenticationConfig string
+	if useAuthenticationConfig {
+		authenticationConfig = fmt.Sprintf(`
+apiVersion: apiserver.config.k8s.io/v1alpha1
+kind: AuthenticationConfiguration
+jwt:
+- issuer:
+    url: %s
+    clientIDs:
+    - %s
+    certificateAuthority: %s
+  claimMappings:
+    username:
+      prefix: %s
+`, oidcServer.URL(), defaultOIDCClientID, base64.StdEncoding.EncodeToString(caCertContent), defaultOIDCUsernamePrefix)
+	}
+
+	apiServer = startTestAPIServerForOIDC(t, oidcServer.URL(), defaultOIDCClientID, caFilePath, authenticationConfig)
```

Comment:
I would split the config more explicitly (applies more than once):

```suggestion
	if useAuthenticationConfig {
		apiServer = startTestAPIServerForOIDC(t, "", "", "", authenticationConfig)
	} else {
		apiServer = startTestAPIServerForOIDC(t, oidcServer.URL(), defaultOIDCClientID, caFilePath, "")
	}
	
```